### DISCUSSION TITLE
Using mc with ble.sh installed

### DISCUSSION THREAD

**Discussion by dkavlakov** [akinomyoga/ble.sh#88] [Q&A]
I noticed that while ble.sh is installed any attempt to use command line in Midnight Commander exits the program. Is there a way to use mc witl ble.sh or to temporary suspend ble.sh when running mc?

**Comment by akinomyoga** [comment]
> I noticed that while ble.sh is installed any attempt to use command line in Midnight Commander exits the program.

Could you describe the detailed behavior? I'm not a heavy user of `mc`, so maybe I miss something, but I can use `mc` with `ble.sh` without problems. Of course, when `mc` is in the foreground, the syntax highlighting and other features of `ble.sh` are not available in the bottom line because that bottom line is provided by another line editor `mc` implements. But if I switch between the shell and `mc` using <kbd>C-o</kbd>, I can still use `ble.sh` without problems in the shell mode.

> Is there a way to use mc witl ble.sh or to temporary suspend ble.sh when running mc?

If you want to turn off `ble.sh` inside `mc`, you can just source `ble.sh` only when the shell is not inside `mc`. For example,

```bash
# bashrc
[[ $- == *i* && ! $MC_SID ]] && source /path/to/ble.sh <some options...>

```

**Reply by dkavlakov** [reply]
> Could you describe the detailed behavior?

- Executing a command exits to the top level shell after the command is executed.
- Pressing ctrl-o places me on blank line, then pressing any key exits to the top level shell

It's fresh Debian 10 install, have bash_competition, fzf and autojump loaded in .bashrc 

> you can just source ble.sh only when the shell is not inside mc

This seems to work nice, thank you

**Reply by akinomyoga** [reply]
OK! So, I could confirm that the behavior is indeed different from my environment. I think some configuration causes the difference of the behavior.

- ~~What are your operating system and its version?~~ -> OK, I saw your edits! > **Debian 10**
- What is your Bash version? You can check it by `echo "$BASH_VERSION ($MACHTYPE)"`
- What is your `ble.sh` version? You can check it by `echo "$BLE_VERSION"`. Ah, I think I should have asked your `ble.sh` version first. ble-0.3 doesn't support `mc`, so `mc` will disable shell features when used with ble-0.3. In this case, you can try the devel version of `ble.sh` in `master` branch.
- What happens if your `~/.bashrc` only contains a single line: `source /path/to/ble.sh`? (Of course, please keep a backup of your original `~/.bashrc`)

**Reply by dkavlakov** [reply]
> What happens if your ~/.bashrc only contains a single line:

OK, this lead me to the issue, it is the last line i added to .bashrc according to installation instructions

`[[ ${BLE_VERSION-} ]] && ble-attach`

If i change it to

`[[ ${BLE_VERSION-} && ! $MC_SID ]] && ble-attach`

mc works fine

**Reply by akinomyoga** [reply]
> ```bash
> [[ ${BLE_VERSION-} && ! $MC_SID ]] && ble-attach
> ```
> mc works fine

Yeah, in this way, `ble.sh` will not be enabled in `mc`, so the problem won't arise. It is OK if you are satisfied with it, but I actually intended to make `ble.sh` work with `mc` properly (even when `ble.sh` is enabled in `mc`). If it would be possible, I want to reproduce and test it in my environment, so if you could provide me the information on the versions of Bash and `ble.sh`, it would be helpful! Thank you!

**Reply by dkavlakov** [reply]
> What is your Bash version?

5.0.3(1)-release (x86_64-pc-linux-gnu)

> What is your ble.sh version?

0.4.0-devel3+3cae6e4

> What happens if your ~/.bashrc only contains a single line: source /path/to/ble.sh

If it is just sourced, without --noattach mc does not work

Tried to use debian stable stock mc package (4.8.22) and own 4.8.26 build.

**Reply by akinomyoga** [reply]
OK. Thanks. Is your Debian 10 the 64 bit version?

**Reply by dkavlakov** [reply]
64 bit base multiarch, used packages are 64 bit.
One more thing i noticed. With the current .bashrc end line (ble-attach only if not in mc) if after running mc i press ctrl-o and execute ble-attach ble starts and works with mc. If i execute ble-attach without ctrl-o (visible mc  interface) it crashes the usual way.
And just to be accurate, it is Devuan 3 (systemd-free base, all else stock Debian 10) with openRC. Can't figure how this can affect the issue, but just in case...

**Reply by akinomyoga** [reply]
Thank you. I've downloaded Debian 10 64 bit image and now installing it in VM. It will take a while.

> If i execute ble-attach without ctrl-o (visible mc interface) it crashes the usual way.

OK, ... I tried the same, but it doesn't reproduce. Everything works fine in my Fedora...

**Reply by akinomyoga** [reply]
I've installed Debian and tried, but it works without the problem.

```console
$ su
# usermod -aG murase sudo
```

Somehow, it didn't work... so I directly added the user name to `/etc/sudoers`.

```console
# vi /etc/sudoers
```

Then

```bash
$ sudo apt install mc git make gawk vim
$ git clone --recursive https://github.com/akinomyoga/ble.sh.git
$ cd ble.sh
$ make
$ make install
$ cd ~
$ mv .bashrc .bashrc.original
$ touch .bashrc
$ vim .bashrc
$ cat .bashrc
source ~/.local/share/blesh/ble.sh
$ mc
```

And it does work without the problem... Do you think there is something different in the above setup? In case it matters, I'm using the default GNOME Terminal.

**Reply by akinomyoga** [reply]
Ah, OK. I think `~/.inputrc` matters. Can you share the content of your `~/.inputrc` with me (if there is `.inputrc` in your home directory)?
